src/cryptographic_primitives/hashing/blake2b512.rs

```diff
@@ -38,8 +42,22 @@ impl Blake {
     }
 
     pub fn result_scalar<E: Curve>(&self) -> Scalar<E> {
-        let n = self.result_bigint();
-        Scalar::from_bigint(&n)
+        let scalar_len = E::Scalar::SCALAR_LENGTH;
```

Comment:
it sucks that we need this code duplication,
I think we should remove those implementations and make the code that requires hashes generic over `Digest` and that way the user can choose their own hash function (and RustCrypto already has a blake2b implementation with simd acceleration: https://github.com/RustCrypto/hashes/tree/master/blake2/src)